ros/src/computing/planning/motion/packages/waypoint_follower/nodes/mpc_follower/lowpass_filter.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2018-2019 Autoware Foundation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mpc_follower/lowpass_filter.h"
+
+Butterworth2dFilter::Butterworth2dFilter(double dt, double f_cutoff_hz)
+{
+  initialize(dt, f_cutoff_hz);
+};
+
+Butterworth2dFilter::~Butterworth2dFilter(){};
+
+void Butterworth2dFilter::initialize(const double &dt, const double &f_cutoff_hz)
+{
+  y1_ = 0.0;
+  y2_ = 0.0;
+  u2_ = 0.0;
+  u1_ = 0.0;
+
+  /* 2d butterworth lowpass filter with bi-linear transformation */
```

Comment:
Since this filter and transformation is very common and simple, I calculated this by hand. So there is no paper for this. I could find the original paper of butterworth filter and write here as a reference, If needed. Is it good?

Reply:
I also checked the result (coefficients) with matlab.

Reply:
OK, no problem. If you have the Matlab coefficients for a couple of cut-off frequencies then those can be added as a test for the function

Reply:
Ok. I added tests.